server/leafnode_test.go

```diff
@@ -4072,3 +4099,942 @@ func TestLeafNodeInterestPropagationDaisychain(t *testing.T) {
+		require_NoError(t, ioutil.WriteFile(confHub, []byte(fmt.Sprintf(tmplHub,
+			sHubUpd2.opts.Port,
+			"enabled",
+			jsEnabled,
+			sHubUpd2.opts.LeafNode.Port,
+			fmt.Sprintf(`default_js_domain: {A:"%s"}`, domain))), 0664))
+
+		if domain != _EMPTY_ {
+			// in case no domain name exists there are no additional guard rails, hence no error
+			// It is the users responsibility to get this edge case right
+			sHubUpd2.Shutdown()
+			sHubUpd2.WaitForShutdown()
+			checkLeafNodeConnectedCount(t, sLeaf, 0)
+			sHubUpd3, err := NewServer(LoadConfig(confHub))
+			sHubUpd3.Shutdown()
+
+			require_Error(t, err)
+			require_Contains(t, err.Error(), `default_js_domain contains account name "A" with enabled JetStream`)
+		}
+	}
+
+	t.Run("with-domain-sys", func(t *testing.T) {
+		test("domain", true)
+	})
+	t.Run("with-domain-nosys", func(t *testing.T) {
+		test("domain", false)
+	})
+	t.Run("no-domain", func(t *testing.T) {
+		test("", true)
+	})
+	t.Run("no-domain", func(t *testing.T) {
+		test("", false)
+	})
+}
+
+func TestLeafNodeJetStreamDefaultDomainJwtExplicit(t *testing.T) {
+	tmplHub := `
+listen: 127.0.0.1:%d
+operator: %s
+system_account: %s
+resolver: MEM
+resolver_preload: {
+	%s:%s
+	%s:%s
+}
+jetstream : disabled
+server_name: HUB
+leafnodes: {
+	listen: 127.0.0.1:%d
+}
+%s
+`
+
+	tmplL := `
+listen: 127.0.0.1:-1
+accounts :{
+    A:{   jetstream: enable, users:[ {user:a1,password:a1}]},
+    SYS:{ users:[ {user:s1,password:s1}]},
+}
+system_account: SYS
+jetstream: { domain: "%s", store_dir: "%s", max_mem: 100Mb, max_file: 100Mb }
+server_name: LEAF
+leafnodes: {
+    remotes:[{url:nats://127.0.0.1:%d, account: A, credentials: %s},
+		     {url:nats://127.0.0.1:%d, account: SYS, credentials: %s}]
+}
+%s
+`
+
+	test := func(domain string) {
+		noDomainFix := ""
+		if domain == _EMPTY_ {
+			noDomainFix = `default_js_domain:{A:""}`
+		}
+
+		sysKp, syspub := createKey(t)
+		sysJwt := encodeClaim(t, jwt.NewAccountClaims(syspub), syspub)
+		sysCreds := newUser(t, sysKp)
+		defer removeFile(t, sysCreds)
+
+		aKp, aPub := createKey(t)
+		aClaim := jwt.NewAccountClaims(aPub)
+		aJwt := encodeClaim(t, aClaim, aPub)
+		aCreds := newUser(t, aKp)
+		defer removeFile(t, aCreds)
+
+		confHub := createConfFile(t, []byte(fmt.Sprintf(tmplHub, -1, ojwt, syspub, syspub, sysJwt, aPub, aJwt, -1, "")))
+		defer removeFile(t, confHub)
+		sHub, _ := RunServerWithConfig(confHub)
+		defer sHub.Shutdown()
+
+		sdLeaf := createDir(t, JetStreamStoreDir)
+		defer os.RemoveAll(sdLeaf)
+		confL := createConfFile(t, []byte(fmt.Sprintf(tmplL,
+			domain,
+			sdLeaf,
+			sHub.opts.LeafNode.Port,
+			aCreds,
+			sHub.opts.LeafNode.Port,
+			sysCreds,
+			noDomainFix)))
+		defer removeFile(t, confL)
+		sLeaf, _ := RunServerWithConfig(confL)
+		defer sLeaf.Shutdown()
+
+		checkLeafNodeConnectedCount(t, sHub, 2)
+		checkLeafNodeConnectedCount(t, sLeaf, 2)
+
+		ncA := natsConnect(t, fmt.Sprintf("nats://127.0.0.1:%d", sHub.opts.Port), createUserCreds(t, nil, aKp))
```

Comment:
I think `defer ncA.Close()` is missing here.

Reply:
fixed